packages/react-server/core/ClientRequest.js

```diff
@@ -15,6 +15,15 @@ class ClientRequest {
 			bundleData,
 			reuseDom,
 		}
+
+		// Chop off the fragment identifier from the url i.e everything from the # to the end of the url
+		// if it exists to make this consistent with ExpressServerRequest.
+		var match = url.match(/([^#]*)/);
+		if (match === null || !match[1]) {
+			console.error("An Invalid url was provided");
+		} else {
+			this._url = match[1];
+		}
```

Comment:
Couple requests for this section:

* I think for the purposes of this PR, I'd prefer not to log anything in the case an "invalid" URL was provided. We could/should validate the URL that we're given, but if we were doing that I'd probably want to `throw` rather than just log, and that's considerably more involved than stripping the hash off the end, so we can put that aside for now.
* I'd like to keep the two assignments to `this._url` in constructor closer together. Feel free to move the first one down into your first if-case.


This doesn't apply to this specific PR, since I'd like to remove the new logging anyway, but for future react-server submissions, we've got a logger set up in react-server that ensures that we get log messages to the appropriate places w/ the appropriate severity levels. In general, rather than `console.error`, we should use that setup (if you search for "logger" in the code, it should be easy to find examples).